coast/TRANSECT.py

```diff
@@ -1,18 +1,313 @@
 from .COAsT import COAsT
-from dask import array
+from scipy.ndimage import convolve1d
+from scipy import interpolate
+import xarray as xr
+import numpy as np
 
 
-class SUBSET(COAsT):
+# =============================================================================
+# The TRANSECT module is a place for code related to transects only
+# =============================================================================
 
-    def __init__(self, domain: COAsT, nemo: COAsT):
-        self.domain_data_obj = domain
-        self.nemo_data_obj = nemo
+class Transect:
+    
+    def __init__(self, domain: COAsT, point_A: tuple, point_B: tuple,
+                 dataset_T: COAsT=None, dataset_U: COAsT=None, dataset_V: COAsT=None):
+        '''
+        Class defining a generic transect type, which is a 3d dataset between a point A and 
+        a point B, with a time dimension, a depth dimension and a transect dimension. The 
+        transect dimension defines the points along the transect.
+        The model Data is subsetted in its entirety along these dimensions.
+        
+        Note that Point A should be of lower latitude than point B.
+        
+        Example usage:
+            point_A = (54,-15)
+            point_B = (56,-12)
+            transect = coast.Transect( domain, point_A, point_B, nemo_t, nemo_u, nemo_v )
+            
 
-    def subset(self, points_a: array, points_b: array):
-        self.dataset = super().transect(self.domain_data_obj, self.nemo_data_obj, points_a, points_b)
+        Parameters
+        ----------
+        domain : DOMAIN, the model domain to extract transect from
+        point_A : tuple, (lat,lon)
+        point_B : tuple, (lat,lon)
+        dataset_T : NEMO, optional, model data on the T grid
+        dataset_U : NEMO, optional, model data on the U grid
+        dataset_V : NEMO, optional, model data on the V grid
+        
+        '''
+        
+        # point A should be of lower latitude than point B
+        if abs(point_B[0]) < abs(point_A[0]):
+            self.point_A = point_B
+            self.point_B = point_A
+        else:
+            self.point_A = point_A
+            self.point_B = point_B
+            
+        # Get points on transect
+        tran_y, tran_x = self.get_transect_indices( domain, self.point_A, self.point_B, "f" )
```

Comment:
tran_y and tran_x are indices, should they be labelled as such: tran_y_ind, tran_x_ind ? Or call them "itran_x" and "itran_y" ?